src/amazonqFeatureDev/client/featureDev.ts

```diff
@@ -156,7 +155,7 @@ export class FeatureDevClient {
             getLogger().error(
                 `${featureName}: failed to execute planning: ${(e as Error).message} RequestId: ${(e as any).requestId}`
             )
-            throw new ToolkitError((e as Error).message, { code: 'GeneratePlanFailed' })
+            throw new ApiError((e as Error).message, 'GeneratePlanFailed', (e as any)?.$metadata?.httpStatusCode ?? 500) // Internal server errors are not returning the http status code
```

Comment:
Is this the only error that doesn't return a status code?

Reply:
All midstream errors are at the moment translated as internal server exception by RTS, I'll update this CR to leave a solution that in the future they may at least return a different error name, so we can get the code form that. I do not expect them to update the midstream issue atm.